dockerfiles/Dockerfile

```diff
@@ -0,0 +1,67 @@
+### This is a full chipyard setup
+
+# BUILD BASE FOR CI
+
+FROM ubuntu:18.04 as base
+ARG CHIPYARD_HASH
+
+MAINTAINER https://groups.google.com/forum/#!forum/chipyard
+
+# Install dependencies for ubuntu-req.sh
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+               curl \
+               git \
+               sudo \
+               ca-certificates \
+               keyboard-configuration \
+               console-setup
+
+# Adds a new user called riscvuser
+RUN groupadd --gid 3434 riscvuser \
+    && useradd --uid 3434 --gid riscvuser --shell /bin/bash --create-home riscvuser \
+    && echo 'riscvuser ALL=NOPASSWD: ALL' >> /etc/sudoers.d/50-riscvuser \
+    && echo 'Defaults    env_keep += "DEBIAN_FRONTEND"' >> /etc/sudoers.d/env_keep
+
+WORKDIR /home/riscvuser
+USER riscvuser
+
+# Update PATH for RISCV toolchain (note: hardcoded for CircleCI)
+ENV RISCV="/home/riscvuser/riscv-tools-install"
+ENV LD_LIBRARY_PATH="$RISCV/lib"
+ENV PATH="$RISCV/bin:$PATH"
+
+# Install Chipyard and run ubuntu-req.sh to install necessary dependencies
+RUN git clone https://github.com/ucb-bar/chipyard.git && \
+        cd chipyard && \
+        git checkout $CHIPYARD_HASH && \
+        ./scripts/ubuntu-req.sh 1>/dev/null && \
+        sudo rm -rf /var/lib/apt/lists/*
```

Comment:
Aren't you already super user here? Can you remove the `sudo`?

Reply:
It feels like I should be able to, but the build runs into permissions issues without the `sudo` for some reason.